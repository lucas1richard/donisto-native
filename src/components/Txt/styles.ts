import { StyleSheet, TextStyle } from 'react-native';
import theme from 'theme/iftheme';

const defaultStyles: TextStyle = {
  fontSize: 18,
  color: theme.black
};

export default StyleSheet.create({
  default: {
    ...defaultStyles
  },
  primary: {
    ...defaultStyles,
    color: theme.primary[500]
  },
  secondary: {
    ...defaultStyles,
    color: theme.secondary[500]
  },
  light: {
    ...defaultStyles,
    color: '#fff'
  },
  muted: {
    ...defaultStyles,
    color: '#bbb'
  },
});
