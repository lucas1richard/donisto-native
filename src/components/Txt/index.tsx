import React from 'react';
import { Text, TextProps } from 'react-native';
import styles from './styles';

interface ITxtProps extends TextProps {
  children: any;
  color?: 'primary'|'secondary'|'default'|'light';
}

const Txt = ({ children, color, style, ...rest }: ITxtProps) => {
  return (
    <Text style={[styles[color], style]} {...rest}>
      {children}
    </Text>
  );
};

Txt.defaultProps = {
  color: 'default'
};

export default Txt;
