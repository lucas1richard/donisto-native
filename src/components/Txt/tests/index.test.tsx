import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Txt, { ITxtProps } from '..';

describe('<Txt />', () => {
  const renderComponent = (props: ITxtProps) => toJson(shallow(<Txt {...props} />));
  it('renders with just text', () => {
    expect(renderComponent({ children: 'mockText' })).toMatchSnapshot();
  });
  it('renders with a color prop', () => {
    expect(renderComponent({
      children: 'mockText',
      color: 'light',
    })).toMatchSnapshot();
  });
});