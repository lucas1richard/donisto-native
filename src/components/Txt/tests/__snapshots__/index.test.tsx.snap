// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Txt /> renders with a color prop 1`] = `
<Component
  style={
    Array [
      Object {
        "color": "#fff",
        "fontSize": 18,
      },
      undefined,
    ]
  }
>
  mockText
</Component>
`;

exports[`<Txt /> renders with just text 1`] = `
<Component
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 18,
      },
      undefined,
    ]
  }
>
  mockText
</Component>
`;
